### PR TITLE
Priority queue with callbacks

### PR DESCRIPTION
**Pull request by jaroslaw-pieszka**
**Description**

Changes proposed in this pull request:

- queue with callbacks in order to use these for metrics gathering,
- queue element wrapped with data tracking time spend in the queue
- extracting with boolean flag to allow multiple consumers 

**Related issue(s)**

#60

### CODE REVIEW

File: internal/syncqueues/priority_queue_with_stats.go [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+	q.elements[q.size] = e
+	q.size++
+	q.siftUp()
+
+	if q.eventHandler != nil && q.eventHandler.OnInsert != nil {
+		q.eventHandler.OnInsert(q.size)
+	}
+}
+
+func (q *SubaccountAwarePriorityQueueWithCallbacks) Extract() QueueElement {
```
Comment (piotrmiskiewicz) [inline internal/syncqueues/priority_queue_with_stats.go:79]: maybe Extract should return a bool which indicates if there was an item or not

**Comment by piotrmiskiewicz** [comment]
do not log as error, when there is no item

**Comment by piotrmiskiewicz** [comment]
do not log as INFO, it is at most DEBUG

File: internal/syncqueues/priority_queue_with_stats.go (hunk 1 above)
Comment (jaroslaw-pieszka) [reply internal/syncqueues/priority_queue_with_stats.go]: Changes pushed

**Review by piotrmiskiewicz — APPROVED** [review]

**Pull request merged**